test/unit-test/ice_api/ice_api_utest.c

```diff
@@ -0,0 +1,31 @@
+/* Unity includes. */
+#include "unity.h"
+#include "catch_assert.h"
+
+/* Standard includes. */
+#include <stdlib.h>
+#include <time.h>
+
+/* API includes. */
+#include "ice_api.h"
+
+/* ===========================  EXTERN VARIABLES  =========================== */
+
+
+/* ==============================  Test Cases  ============================== */
+
+/**
+ * @brief Validate ICE Init fail functionality for Bad Parameters.
+ */
+void test_iceInit_BadParams( void )
+{
+    IceResult_t result;
+
+    result = Ice_Init( NULL,
+                       NULL );
```

Comment:
Since the API checks the conditions using OR operations, it would be better to check out all the cases using separate tests